apollo-router/src/uplink/license_enforcement.rs

```diff
@@ -82,7 +85,69 @@ where
 #[derive(Debug)]
 pub(crate) struct LicenseEnforcementReport {
     restricted_config_in_use: Vec<ConfigurationRestriction>,
-    restricted_schema_in_use: Vec<SchemaRestriction>,
+    restricted_schema_in_use: Vec<SchemaViolation>,
+}
+
+#[derive(Debug)]
+struct ParsedLinkSpec {
+    spec_name: String,
+    version: semver::Version,
+    base_url: String,
+    imported_as: Option<String>,
+    url: String,
+}
+
+impl ParsedLinkSpec {
+    fn from_link_directive(link_directive: &Node<Directive>) -> Option<ParsedLinkSpec> {
+        link_directive
+            .argument_by_name(LINK_URL_ARGUMENT)
+            .and_then(|value| {
+                let Some(url) = value.as_str() else {
+                    return None;
+                };
+                // capture url + spec name, spec name, and version from a link spec url
+                let re = Regex::new(r"^(.*/([^/]+))/v(\d+\.\d+)$").unwrap();
+                let Some(captures) = re.captures(url) else {
+                    return None;
+                };
+
+                let full_url = captures.get(0).unwrap().as_str().to_string();
```

Comment:
with this unwrap the router would panic on a schema with an invalid spec URL. That function should probably return a `Result<Option<ParsedLinkSpec>, SomeError>`, to account for both cases of no spec found, or parsing error